consensus/event_loop.go

```diff
@@ -521,6 +550,23 @@ func (fc *FastHotStuffEventLoop) hasTimedOutForView(publicKey *bls.PublicKey, vi
 	return ok
 }
 
+func (fc *FastHotStuffEventLoop) getBlockAndValidatorSetByHash(blockHash BlockHash) (
+	bool, Block, []Validator, map[string]Validator,
+) {
+	// A linear search here is fine. The safeBlocks slice is expected to be extremely small as it represents the
+	// number of uncommitted blocks in the blockchain. During steady stake, it will have a size of 3 blocks
+	// (one committed, two uncommitted). In the worse case, where the network has an unlucky series of
+	// timeout -> block -> timeout -> block,... it can still be expected to have < 10 blocks.
+	blockHashValue := blockHash.GetValue()
+	for _, block := range fc.safeBlocks {
```

Comment:
Could we use your `collections.Any` here?

Reply:
We want to extract the block and validator set at this block height, so a simple loop is straightfoward here. Not ideal to use collections.Any since that's only meant to return true/false based on a condition.